flytekit/annotated/launch_plan.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import annotations
+
+from typing import Dict, Any, List
+
+from flytekit import logger
+from flytekit.annotated import workflow
+from flytekit.annotated.context_manager import FlyteContext
+from flytekit.annotated.interface import (
+    transform_inputs_to_parameters,
+    Interface,
+)
+from flytekit.annotated.type_engine import TypeEngine
+from flytekit.common.launch_plan import SdkLaunchPlan
+from flytekit.models import common as _common_models
+from flytekit.models import interface as _interface_models
+from flytekit.models import launch_plan as _launch_plan_models
+from flytekit.models import literals as _literal_models
+from flytekit.models import schedule as _schedule_model
+from flytekit.models.core import identifier as _identifier_model
+
+
+class LaunchPlan(object):
+    # The reason we cache is simply because users may get the default launch plan twice for a single Workflow. We
```

Comment:
how is that possible, Flyteadmin will reject it right?

Reply:
Admin will be fine yeah, i just mean locally.  I think it makes more sense to return the same object if for instance the same launch plan is used in two different parent workflows.